### ISSUE TITLE
NotFound when hitting templates API endpoint to generate repo from certain templates

### ISSUE THREAD

**Issue by EricPickup** [education/classroom#2202]
We've been investigating this issue for a while without success.

Haystack: https://haystack.githubapp.com/classroom/rollups/5b37ca2f1b25097be9b95d0d34f3e299

Certain template repos will return a NotFound error when hitting the `/repositories/:id/generate` or `/repos/owner/name/generate` endpoints to create a repository from a template repo. Most of these failures are with private repos so we don't have many examples. We've been using this repo: https://github.com/swe4s/unit-test.

The repos that fail will fail 100% of the time.

We're able to use the template through the web UI (which uses GraphQL) which tells us the issue is with the REST adaptation.

We discovered that when using curl to hit the endpoint using an OAuth token we receive the NotFound error, but using a personal access token will succeed. We're able to use both OAuth and personal access tokens for other template repos that are working fine.

We cannot reproduce the error through a local instance of GitHub. It only fails in production.

```
curl -H "Content-Type: application/json" -d '{"name":"new-repo-name"}' -X POST -H "Authorization: token <your token>" -H "Accept: application/vnd.github.baptiste-preview" https://api.github.com/repos/swe4s/unit-test
```
returns
```
{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3"
}
```

but using any other template repo with the same token succeeds.

**Comment by jeffrafter** [comment]
We're now able to reproduce this issue. The problem happens when the organization that owns the repository (even a public repository) restricts third party access:

![](https://rpl.cat/uploads/VoHmmN0KLE3YHIfSafoSMb5WJkeNl7Kpw4PoQnSu9oM/public.png)

**Comment by EricPickup** [comment]
After a lot of investigation we determined that #2222 was the biggest source of these errors.

https://github.com/education/classroom/issues/2202#issuecomment-519300046 was still the case with one or two orgs and we've added proper error messages to help teachers out with this, and stopped reporting the errors to Haystack since this is expected behavior when teachers modify their org's third-party access. We'll add some info in the help docs to fix this easily.
